### PR TITLE
feat: add a command for updating and signing targets of specified type

### PR DESCRIPTION
**Pull request by renatav**
## Description (e.g. "Related to ...", etc.)

Added a command which updates target files corresponding to target repositories specified through the target type input parameter. Automatically sign and commit

`taf targets update-and-sign-targets law --target-type html --target-type xml --keystore ..\keystore-dev\keystore-dev`

_Please replace this description with a concise description of this Pull Request._

## Code review checklist (for code reviewer to complete)

- [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [ ] Title summarizes what is changing
- [ ] Commit messages are meaningful (see [this][commit messages] for details)
- [ ] Tests have been included and/or updated, as appropriate
- [ ] Docstrings have been included and/or updated, as appropriate
- [ ] Changelog has been updated, as needed (see [CHANGELOG.md][changelog])

[changelog]: https://github.com/openlawlibrary/taf/blob/master/CHANGELOG.md
[commit messages]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: taf/developer_tool.py [hunk 1]
```diff
@@ -1272,6 +1282,61 @@ def setup_test_yubikey(key_path=None):
...
+    for target_type in target_types:
+        try:
+            target_name = repositoriesdb.get_repositories_paths_by_custom_data(
+                auth_repo, type=target_type
+            )[0]
+        except Exception:
+            print(
+                f"Skipping {target_type}. Type listed in repositories.json as a target repository"
+            )
+            continue
```
Comment (n-dusan) [inline taf/developer_tool.py:1329]: Should we raise an error when we specify `target_type` and it's not found, or do we want to continue and sign other targets? If we're skipping `target_types` which don't exist in `repositories.json`, would it be simpler to just sign the types which exist in repositories.json (and not need to specify types in the cli)?

File: taf/developer_tool.py [hunk 2]
```diff
@@ -1272,6 +1282,61 @@ def setup_test_yubikey(key_path=None):
...
+    repositoriesdb.load_repositories(auth_repo)
+    signed_targets = []
+    for target_type in target_types:
+        try:
+            target_name = repositoriesdb.get_repositories_paths_by_custom_data(
+                auth_repo, type=target_type
+            )[0]
+        except Exception:
+            print(
+                f"Skipping {target_type}. Type listed in repositories.json as a target repository"
```
Comment (n-dusan) [inline taf/developer_tool.py:1327]:
```suggestion
                f"Skipping {target_type}. Type not listed in repositories.json as a target repository"
```

**Review by n-dusan — CHANGES_REQUESTED** [review]

File: taf/tools/metadata/__init__.py [hunk 3]
```diff
@@ -22,6 +22,8 @@ def metadata():
                   "keys or a path to a json file which stores the needed information")
     @click.option("--scheme", default=DEFAULT_RSA_SIGNATURE_SCHEME, help="A signature scheme "
                   "used for signing")
+    @click.option("--no-commit", is_flag=True, default=False, help="Indicates if the changes should not be "
+                  "committed automatically")
```
[the remark this answers is not in the document]
Comment (n-dusan) [reply taf/tools/metadata/__init__.py:26]: Yes, makes sense to not add this option

File: taf/developer_tool.py (hunk 1 above)
Comment (n-dusan) [reply taf/developer_tool.py:1329]: > Maybe we should check if they all exist before updating anything? In this case, if you make a typo, you'd have to run the command again

Okay, makes sense

> We already have a command which updates everything listed in repositories.json, but it proved to be quite problematic. 

Which command is that?

File: taf/tools/metadata/__init__.py (hunk 3 above)
Comment (n-dusan) [inline taf/tools/metadata/__init__.py:26]: Did we forget to remove this?

File: taf/developer_tool.py (hunk 1 above)
Comment (renatav) [reply taf/developer_tool.py]: uodate-repos-from-repositories-json

**Review by n-dusan — APPROVED** [review]

**Pull request merged**